### ISSUE TITLE
Optional after character classes fails to match.

### ISSUE THREAD

**Issue by raffalevy** [maciejhirsz/logos#114]
Logos fails to match the string `"F0"` to the regex `[0-9A-F][0-9A-F]a?`, as in the example below:

```rust

// 0.11.0-rc2
use logos::Logos;
use crate::LogosToken::Tok;

fn main() {
    let mut lexer = LogosToken::lexer("F0");
    assert_eq!(lexer.next().unwrap(), Tok); // fails
}

#[derive(Logos, Debug, PartialEq)]
enum LogosToken {
    #[error]
    Error,

    #[regex("[0-9A-F][0-9A-F]a?")]
    Tok,
}
```

If the regex is changed to `[A-F][0-9]a?`, it matches the string `"F0"` as expected. The regex `[0-9A-F][0-9A-F]a?` does match `"F0a"`.

**Comment by maciejhirsz** [comment]
Thanks for reporting. It ended up being an issue with optional character in a token at the tail of the input, so unrelated to being after a class as such.

Released 0.11.0-rc3 with the fix.

**Comment by raffalevy** [comment]
This fixed the issue for me. Thank you!
